website/docs/dev_colorspace.md

```diff
@@ -0,0 +1,113 @@
+---
+id: dev_colorspace
+title: Colorspace Management and Distribution
+sidebar_label: Colorspace
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Introduction
+Defines the distribution of colors and OCIO config during publishing. Once colorspace data are captured and integrated into representation loaders could use them for loading image and video data with correct colorspace.
+
+:::warning Color Management (ImageIO)
+Settings schema is requred for any host or module which is processing pixel data.
+:::
+
+## Data model
+The *colorspaceData* are stored at root of representation dictionary during publishing. Once they are integrated into representation db document they are stored as *representation_doc.data["colorspaceData"]*
+
+### Keys
+- **colorspace** - string value used in other publish plugins and loaders
+- **configData** - storing two versions of path.
+  - **path** - is formated and with baked platform root. It is used for posible need to find out where we were sourcing color config during publishing.
+  - **template** - unformated tempate resolved from settings. It is used for other plugins targeted to remote publish which could be processed at different platform.
+
+### Example
+    {
+        "colorspace": "linear",
+        "configData": {
+            "path": "/abs/path/to/config.ocio",
+            "template": "{project[root]}/path/to/config.ocio"
+        }
+    }
+
+
+## How to integrate it into a host
+1. Each host setting schema should have following shemas. Ideally at top of all categories so it mimic already defined order at other hosts.
+```json
+{
+    "key": "imageio",
+    "type": "dict",
+    "label": "Color Management (ImageIO)",
+    "is_group": true,
+    "children": [
+        {
+            "type": "schema",
+            "name": "schema_imageio_config"
+        },
+        {
+            "type": "schema",
+            "name": "schema_imageio_file_rules"
+        }
+
+    ]
+}
+```
+
+2. Use any mechanism to set OCIO config to host app resolved from `openpype\pipeline\colorspace.py:get_imageio_config`
+	-	either set OCIO environment during host launching via pre-launch hook
+	- or to set workfile ocio config path if host api is available
+
+3. Each pixle related exporter plugins has to use parent class `openpype\pipeline\publish\publish_plugins.py:ExtractorColormanaged` and use it similarly as it is already implemented here `openpype\hosts\nuke\plugins\publish\extract_render_local.py`
+- **get_colorspace_settings**: is solving all settings for the host context
+- **set_representation_colorspace**: is adding colorspaceData to representation. If the colorspace is known then it is added directly to the representation with resolved config path.
```

Comment:
```suggestion
3. Each Extractor exporting pixel data (e.g. image or video) has to use parent class `openpype.pipeline.publish.publish_plugins.ExtractorColormanaged` and use `self.set_representation_colorspace` on the representations to be integrated.

The **set_representation_colorspace** method adds `colorspaceData` to the representation. If the `colorspace` passed is not `None` then it is added directly to the representation with resolved config path otherwise a color space is assumed using the configured file rules. If no file rule matches the `colorspaceData` is **not** added to the representation.

An example implementation can be found here: `openpype\hosts\nuke\plugins\publish\extract_render_local.py`

```